loky/backend/semlock.py

```diff
@@ -49,22 +54,43 @@ def get_ident():
 
 
 if sys.version_info[:2] < (3, 3):
-    FileExistsError = OSError
+    class FileExistsError(OSError):
+        pass
+
+    class FileNotFoundError(OSError):
+        pass
 
 
 def sem_unlink(name):
     if pthread.sem_unlink(name) < 0:
         raiseFromErrno()
 
 
-def _sem_open(name, o_flag, perm=None, value=None):
-    if perm is None:
-        assert value is None, "Wrong number of argument (either 2 or 4)"
-        return pthread.sem_open(ctypes.c_char_p(name), o_flag)
+def _sem_open(name, value=None):
+    """ Construct or retrieve a semaphore with the given name
+
+    If value is None, try to retrieve an existing named semaphore.
+    Else create a new semaphore with the given value
+    """
+    if value is None:
+        handle = pthread.sem_open(ctypes.c_char_p(name), 0)
     else:
-        assert value is not None, "Wrong number of argument (either 2 or 4)"
-        return pthread.sem_open(ctypes.c_char_p(name), ctypes.c_int(o_flag),
-                                ctypes.c_int(perm), ctypes.c_int(value))
+        handle = pthread.sem_open(ctypes.c_char_p(name), SEM_OFLAG, SEM_PERM,
+                                  ctypes.c_int(value))
+
+    if handle == SEM_FAILURE:
+        e = ctypes.get_errno()
+        if e == errno.EEXIST:
+            raise FileExistsError('cannot find name for semaphore')
```

Comment:
This would be cleaner to have a proper exception name: it would make the code easier to understand even if the user never gets that exception. +1 for merging once this comment is addressed.